core/services/fluxmonitor/flux_monitor.go

```diff
@@ -455,8 +475,75 @@ func (p *PollingDeviationChecker) consume(ctx context.Context, roundSubscription
 	}
 }
 
-func (p *PollingDeviationChecker) pollIfEligible(client eth.Client, threshold float64) bool {
-	open, err := p.isEligibleToPoll(client)
+func (p *PollingDeviationChecker) respondToLog(log interface{}) (roundStarted bool, roundEnded bool, _ error) {
+	switch log := log.(type) {
+	case contracts.LogNewRound:
+		return true, false, p.respondToNewRoundLog(log)
+	case contracts.LogRoundDetailsUpdated:
+		return false, false, p.respondToRoundDetailsUpdatedLog(log)
+	case contracts.LogAnswerUpdated:
+		return false, true, p.respondToAnswerUpdatedLog(log)
+	default:
+		panic("got unknown log")
+	}
+}
+
+func (p *PollingDeviationChecker) respondToAnswerUpdatedLog(log contracts.LogAnswerUpdated) error {
+	return nil
+}
+
+func (p *PollingDeviationChecker) respondToRoundDetailsUpdatedLog(log contracts.LogRoundDetailsUpdated) error {
+	p.roundTimeout = time.Duration(log.Timeout.Int64()) * time.Second
+	return nil
+}
+
+// respondToNewRoundLog takes the round broadcasted in the log event, and responds
+// on-chain with an updated price.
+// Only invoked by the CSP consumer on the single goroutine for thread safety.
+func (p *PollingDeviationChecker) respondToNewRoundLog(log contracts.LogNewRound) error {
+	requestedRound := log.RoundId
+
+	jobSpecID := p.initr.JobSpecID.String()
+	promSetBigInt(promFMSeenRound.WithLabelValues(jobSpecID), requestedRound)
+
+	// skip if requested is not greater than current.
```

Comment:
If we're gonna have a comment it might as well explain why as well as what.

Reply:
This is improved by the next PR in this family of changes.